### PR TITLE
CI Test Migrate from dep to go modules

### PR DESCRIPTION
**Pull request by div-co**
## Motivation

*As `dep` is deprecated a migration to go modules `go mod` is required. [related issue](https://github.com/Peripli/service-manager/issues/627).*

## Approach

_By modifying relevant code to support `go mod` command instead of `dep`._

## Pull Request status

* [x] Initial implementation
* [ ] Refactoring
* [ ] Unit tests
* [ ] Integration tests

Pull Request should be ready for merge to master.

## Third-party code

No changes.

[1]: https://github.com/Peripli/service-manager/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -25,14 +25,16 @@ jobs:
       with:
         path: ${{ env.GOPATH }}/src/github.com/${{ github.repository }}
         
-    - name: Get dependencies
-      working-directory: ${{ env.GOPATH }}/src/github.com/${{ github.repository }}
-      run: |
-        curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
+    #- name: Get dependencies
```
Comment (alexwo) [inline .github/workflows/go.yml:28]: we should remove this comments

File: .github/workflows/go.yml [hunk 2]
```diff
@@ -69,10 +71,10 @@ jobs:
...
         path: ${{ env.GOPATH }}/src/github.com/${{ github.repository }}
 
-    - name: Get dependencies
-      working-directory: ${{ env.GOPATH }}/src/github.com/${{ github.repository }}
-      run: |
-        curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
+    #- name: Get dependencies
+    #  working-directory: ${{ env.GOPATH }}/src/github.com/${{ github.repository }}
+    #  run: |
+    #    curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
```
Comment (alexwo) [inline .github/workflows/go.yml:77]: let's remove the comments?

File: .github/workflows/go.yml [hunk 3]
```diff
@@ -113,10 +115,10 @@ jobs:
...
```
Comment (alexwo) [inline .github/workflows/go.yml:121]: let's remove the comments?

File: .github/workflows/go.yml [hunk 4]
```diff
@@ -157,10 +159,10 @@ jobs:
       with:
         path: ${{ env.GOPATH }}/src/github.com/${{ github.repository }}
 
-    - name: Get dependencies
-      working-directory: ${{ env.GOPATH }}/src/github.com/${{ github.repository }}
-      run: |
-        curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
+    #- name: Get dependencies
+    #  working-directory: ${{ env.GOPATH }}/src/github.com/${{ github.repository }}
+    #  run: |
```
Comment (alexwo) [inline .github/workflows/go.yml:164]: let's remove the comments?

File: .github/workflows/go.yml [hunk 5]
```diff
@@ -200,11 +202,16 @@ jobs:
...
-
+    
     - name: Get dependencies
       working-directory: ${{ env.GOPATH }}/src/github.com/${{ github.repository }}
       run: |
-        curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
+        go mod vendor
+
+    #- name: Get dependencies
+    #  working-directory: ${{ env.GOPATH }}/src/github.com/${{ github.repository }}
```
Comment (alexwo) [inline .github/workflows/go.yml:212]: let's remove the comments?

Comment (alexwo) [inline .github/workflows/go.yml:212]: let's remove the comments?

File: Dockerfile [hunk 6]
```diff
@@ -7,14 +7,16 @@ FROM golang:1.12.7-alpine3.10 AS builder
 
 # We need so that dep can fetch it's dependencies
 RUN apk --no-cache add git
-RUN go get github.com/golang/dep/cmd/dep
+#RUN go get github.com/golang/dep/cmd/dep
```
Comment (alexwo) [inline Dockerfile:10]: let's remove the comments?

File: Makefile [hunk 7]
```diff
@@ -16,7 +16,7 @@ all: build test-unit ## Default target that builds SM and runs unit-tests
 
 GO 					?= go
 GOFMT 				?= gofmt
-BINDIR 				?= bin
+BINDIR 				?= ${GOPATH}/bin
```
Comment (alexwo) [inline Makefile:19]: is this change needed?

File: Makefile [hunk 8]
```diff
@@ -74,27 +74,23 @@ COUNTERFEITER   ?= "v6.0.2"
...
-	@gometalinter --install
+	@echo "Installing gometalinter to ..."
+	@cd ${GOPATH}/src;\
+		go get -u github.com/alecthomas/gometalinter;\
+		cd ${GOPATH}/src/github.com/alecthomas/gometalinter;\
+		go install;\
+		gometalinter -i -u
 endif
+# golangci-lint replacing depricated gometalinter implementation will be postponed
+#ifeq ($(shell which golangci-lint),)
```
Comment (alexwo) [inline Makefile:90]: let's remove comments

**Comment by alexwo** [comment]
@div-co please fill the PR template

File: Makefile (hunk 8 above)
Comment (div-co) [reply Makefile]: Done

File: Dockerfile (hunk 6 above)
Comment (div-co) [reply Dockerfile]: Removed

File: .github/workflows/go.yml (hunk 5 above)
Comment (div-co) [reply .github/workflows/go.yml]: Removed

Comment (div-co) [reply .github/workflows/go.yml]: Done

File: .github/workflows/go.yml (hunk 4 above)
Comment (div-co) [reply .github/workflows/go.yml]: Removed

File: .github/workflows/go.yml (hunk 1 above)
Comment (div-co) [reply .github/workflows/go.yml]: Removed

File: Makefile (hunk 7 above)
Comment (div-co) [reply Makefile]: GOPATH not required here anymore

**Review by alexwo — APPROVED** [review]
